biopandas/mmcif/mmcif_parser.py

```diff
@@ -0,0 +1,444 @@
+# The CIF parser relieves heavily on the parser from PDBJapan (author: Gert-Jan Bekker).
+# Python CIF parser: https://gitlab.com/pdbjapan/tools/cif-parsers
+# License: MIT
+#   See https://gitlab.com/pdbjapan/tools/cif-parsers/blob/master/LICENSE
+import gzip
+import re
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+try:
+    str.partition
+
+    def partitionString(string, sep):
+        return string.partition(sep)
+
+
+except:
+
+    def partitionString(string, sep):
+        tmp = string.split(sep)
+        return [tmp.pop(0), sep, sep.join(tmp)]
+
+
+class _loop:
+    def __init__(self, parserObj):
+        self.parserObj = parserObj
+        self.length = 0
+        self.refID = -1
+
+        self.refList = []
+
+        self.namesDefined = False
+
+    def addName(self, name):
+        catName = type(name) == str and partitionString(
+            name, ".") or ["", "", ""]
+        if catName[1]:
+            if not catName[0] in self.parserObj.currentTarget[-2]:
+                self.parserObj.currentTarget[-2][catName[0]] = {}
+            if not catName[2] in self.parserObj.currentTarget[-2][catName[0]]:
+                self.parserObj.currentTarget[-2][catName[0]][catName[2]] = []
+            self.refList.append(
+                self.parserObj.currentTarget[-2][catName[0]][catName[2]]
+            )
+        else:
+            if not catName[0] in self.parserObj.currentTarget[-2]:
+                self.parserObj.currentTarget[-2][catName[0]] = []
+            self.refList.append(self.parserObj.currentTarget[-2][catName[0]])
+        self.length = len(self.refList)
+
+    def pushValue(self, value):
```

Comment:
could we change all the methods to the Python function syntax, e.g., `push_value` etc.?

Reply:
Yep! 🐫